src/Nox.Types.Abstractions/Enums/NoxType.cs

```diff
@@ -10,22 +10,50 @@ public enum NoxType : uint
     Object = 2524865811,
 
     // Compound Types - requires multiple fields to persist
+
+    [CompoundType]
+    [CompoundComponent("Start", typeof(DateTime))]
+    [CompoundComponent("End", typeof(DateTime))]
+    DateTimeRange = 3837929056,
+
     [CompoundType]
     // Entity depends on the concrete instance and if it supports multiple keys or not!
     [CompoundComponent("Type", typeof(string))]
     [CompoundComponent("Id", typeof(uint))]
     Entity = 2276495181,
 
     [CompoundType]
-    [CompoundComponent("Latitude",typeof(double))]
-    [CompoundComponent("Longitude",typeof(double))]
+    [CompoundComponent("Url", typeof(string))]
+    [CompoundComponent("PrettyName", typeof(string))]
+    [CompoundComponent("SizeInBytes", typeof(uint))]
+    File = 612041382,
+
+    [CompoundType]
+    [CompoundComponent("HashText", typeof(string))]
+    [CompoundComponent("Salt", typeof(string))]
+    HashedText = 3656553818,
+
+    [CompoundType]
+    [CompoundComponent("Url", typeof(string))]
+    [CompoundComponent("PrettyName", typeof(string))]
+    [CompoundComponent("SizeInBytes", typeof(uint))]
+    Image = 3650429592,
+
+    [CompoundType]
+    [CompoundComponent("Latitude",typeof(decimal))]
```

Comment:
why change from double to Decimal?

Reply:
he OData database context gives error with double:
can't parse JSON. Raw result:

{"@odata.context":"[https://localhost:7024/api/$metadata#AllNoxTypes","value":[{"statusCode":500,"message":"Error](https://localhost:7024/api/$metadata#AllNoxTypes%22,%22value%22:%5B%7B%22statusCode%22:500,%22message%22:%22Error) occured during request: /api/AllNoxTypes.Error: Unable to cast object of type \u0027System.Decimal\u0027 to type \u0027System.Double\u0027."}

![261013688-93c9ab93-9eeb-4c77-b91a-db463f87b879](https://github.com/NoxOrg/Nox.Generator/assets/136477073/b29eb2b7-6eda-4c5e-a323-96072fd464ad)

![261013793-1429e133-e10a-486a-ac3b-8029a6bb4c78](https://github.com/NoxOrg/Nox.Generator/assets/136477073/a6587efc-5770-4f9f-9be2-1cb77c2a6951)

But works well with Decimal.

Reply:
main has the LatLong already set working properly , this change is not required, must be another problem